packages/flutter/lib/src/widgets/primary_scroll_controller.dart

```diff
@@ -56,10 +58,23 @@ class PrimaryScrollController extends InheritedWidget {
   /// Returns the [ScrollController] most closely associated with the given
   /// context.
   ///
+  /// If `rootState` is set to true, the [ScrollController] from the furthest
+  /// Scaffold instead. Useful for using nested [Navigator].
+  ///
   /// Returns null if there is no [ScrollController] associated with the given
   /// context.
-  static ScrollController? of(BuildContext context) {
-    final PrimaryScrollController? result = context.dependOnInheritedWidgetOfExactType<PrimaryScrollController>();
+    static ScrollController? of(
+    BuildContext context, {
+    bool rootState = false,
+  }) {
+    BuildContext ctx;
+    if (rootState) {
+      ctx = context.findRootAncestorStateOfType<ScaffoldState>()?.context ?? context;
```

Comment:
Why is the Scaffold relevant here?

Reply:
This would not actually solve the problem, wouldn't you need to change the `scaffold.dart` to fix the issue? Also, what about folks that do not want to use the root navigator?

Reply:
> This would not actually solve the problem, wouldn't you need to change the `scaffold.dart` to fix the issue? Also, what about folks that do not want to use the root navigator?

I think I should change `routes.dart`.

Here
https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter/lib/src/widgets/routes.dart#L729

and here
https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter/lib/src/widgets/routes.dart#L810

If you don't want to use root navigator, PrimaryScrollController returns current navigator's PrimaryScrollController.
If want to use root navigator, PrimaryScrollController returns root navigator's PrimaryScrollController.

Reply:
Right, I think I follow, so how would the Scaffold decide which one it wanted - since the Scaffold is the one that calls on the controller, not the user?